File: master_of_plow_CPP_SDK/src-gen/nevonex/fcal/RECEIVE_SFT_INFO_T_R.cpp
/*
 * nevonex/fcal/RECEIVE_SFT_INFO_T_R.cpp
 * Copyright (c) Robert Bosch GmbH. All rights reserved.
 */

#include "RECEIVE_SFT_INFO_T_R.hpp"
#include <ecore/EObject.hpp>
#include <ecore/EClass.hpp>
#include "nevonex/fcal/FcalPackage.hpp"
#include <ecorecpp/mapping.hpp>
#include <algorithm>
#include <nevonex/fcb/SubscriberEnum.hpp>
#include <nevonex-fcal-platform/log/Logger.hpp>

#if BOOST_VERSION >= 106501
#define BOOST_STACKTRACE_HEADERS_FOUND
#include <boost/stacktrace.hpp>
#endif

/*PROTECTED REGION ID(RECEIVE_SFT_INFO_T_R.cpp) START*/
// Please, enable the protected region if you add manually written code.
// To do this, add the keyword ENABLED before START.
/*PROTECTED REGION END*/

using namespace ::nevonex::fcal;
using namespace ::nevonex::log;

// Default constructor
RECEIVE_SFT_INFO_T_R::RECEIVE_SFT_INFO_T_R() : m_tRZ_SFT_SIG1_V(0.0f), m_tRZ_SFT_SIG2_V(0.0f), m_tRZ_SFT_DIAG(0), m_tRZ_SFT_AUTO(0), m_tRZ_SFT_STATE(0)
{

    /*PROTECTED REGION ID(RECEIVE_SFT_INFO_T_R__RECEIVE_SFT_INFO_T_R) START*/
    // Please, enable the protected region if you add manually written code.
    // To do this, add the keyword ENABLED before START.
    /*PROTECTED REGION END*/

}

RECEIVE_SFT_INFO_T_R::~RECEIVE_SFT_INFO_T_R()
{

    /*PROTECTED REGION ID(RECEIVE_SFT_INFO_T_R__Destructor) START*/
    // Please, enable the protected region if you add manually written code.
    // To do this, add the keyword ENABLED before START.
    /*PROTECTED REGION END*/

}

// Attributes

::ecore::EFloat RECEIVE_SFT_INFO_T_R::getTRZ_SFT_SIG1_V() const
{
    return m_tRZ_SFT_SIG1_V;
}

void RECEIVE_SFT_INFO_T_R::setTRZ_SFT_SIG1_V(::ecore::EFloat _tRZ_SFT_SIG1_V)
{

    ::ecore::EFloat _old_tRZ_SFT_SIG1_V = m_tRZ_SFT_SIG1_V;
    m_tRZ_SFT_SIG1_V = _tRZ_SFT_SIG1_V;

}

::ecore::EFloat RECEIVE_SFT_INFO_T_R::getTRZ_SFT_SIG2_V() const
{
    return m_tRZ_SFT_SIG2_V;
}

void RECEIVE_SFT_INFO_T_R::setTRZ_SFT_SIG2_V(::ecore::EFloat _tRZ_SFT_SIG2_V)
{

    ::ecore::EFloat _old_tRZ_SFT_SIG2_V = m_tRZ_SFT_SIG2_V;
    m_tRZ_SFT_SIG2_V = _tRZ_SFT_SIG2_V;

}

::ecore::EInt RECEIVE_SFT_INFO_T_R::getTRZ_SFT_DIAG() const
{
    return m_tRZ_SFT_DIAG;
}

void RECEIVE_SFT_INFO_T_R::setTRZ_SFT_DIAG(::ecore::EInt _tRZ_SFT_DIAG)
{

    ::ecore::EInt _old_tRZ_SFT_DIAG = m_tRZ_SFT_DIAG;
    m_tRZ_SFT_DIAG = _tRZ_SFT_DIAG;

}

::ecore::EInt RECEIVE_SFT_INFO_T_R::getTRZ_SFT_AUTO() const
{
    return m_tRZ_SFT_AUTO;
}

void RECEIVE_SFT_INFO_T_R::setTRZ_SFT_AUTO(::ecore::EInt _tRZ_SFT_AUTO)
{

    ::ecore::EInt _old_tRZ_SFT_AUTO = m_tRZ_SFT_AUTO;
    m_tRZ_SFT_AUTO = _tRZ_SFT_AUTO;

}

::ecore::EInt RECEIVE_SFT_INFO_T_R::getTRZ_SFT_STATE() const
{
    return m_tRZ_SFT_STATE;
}

void RECEIVE_SFT_INFO_T_R::setTRZ_SFT_STATE(::ecore::EInt _tRZ_SFT_STATE)
{

    ::ecore::EInt _old_tRZ_SFT_STATE = m_tRZ_SFT_STATE;
    m_tRZ_SFT_STATE = _tRZ_SFT_STATE;

}

// References

File: com.bosch.fsp.master_of_plow.gen/src/com/bosch/nevonex/fcal/impl/RECEIVE_SFT_INFO_T_R.java
/**
Copyright (c) Robert Bosch GmbH. All rights reserved.
*/
package com.bosch.nevonex.fcal.impl;

import com.bosch.nevonex.fcal.IRECEIVE_SFT_INFO_T_R;

import org.eclipse.emf.ecore.EClass;
import org.eclipse.emf.ecore.EObject;
import org.eclipse.emf.ecore.InternalEObject;

import org.eclipse.emf.ecore.impl.EObjectImpl;

/**
 * @generated
 */
public class RECEIVE_SFT_INFO_T_R extends EObjectImpl implements IRECEIVE_SFT_INFO_T_R {
	protected float tRZ_SFT_SIG1_V = 0.0f;
	protected float tRZ_SFT_SIG2_V = 0.0f;
	protected int tRZ_SFT_DIAG = 0;
	protected int tRZ_SFT_AUTO = 0;
	protected int tRZ_SFT_STATE = 0;

	protected RECEIVE_SFT_INFO_T_R() {
		super();
	}

	@Override
	protected EClass eStaticClass() {
		return FcalPackage.Literals.RECEIVE_SFT_INFO_T_R;
	}

	public float getTRZ_SFT_SIG1_V() {
		return tRZ_SFT_SIG1_V;
	}

	public void setTRZ_SFT_SIG1_V(float newTRZ_SFT_SIG1_V) {
		tRZ_SFT_SIG1_V = newTRZ_SFT_SIG1_V;
	}

	public float getTRZ_SFT_SIG2_V() {
		return tRZ_SFT_SIG2_V;
	}

	public void setTRZ_SFT_SIG2_V(float newTRZ_SFT_SIG2_V) {
		tRZ_SFT_SIG2_V = newTRZ_SFT_SIG2_V;
	}

	public int getTRZ_SFT_DIAG() {
		return tRZ_SFT_DIAG;
	}

	public void setTRZ_SFT_DIAG(int newTRZ_SFT_DIAG) {
		tRZ_SFT_DIAG = newTRZ_SFT_DIAG;
	}

	public int getTRZ_SFT_AUTO() {
		return tRZ_SFT_AUTO;
	}

	public void setTRZ_SFT_AUTO(int newTRZ_SFT_AUTO) {
		tRZ_SFT_AUTO = newTRZ_SFT_AUTO;
	}

	public int getTRZ_SFT_STATE() {
		return tRZ_SFT_STATE;
	}

	public void setTRZ_SFT_STATE(int newTRZ_SFT_STATE) {
		tRZ_SFT_STATE = newTRZ_SFT_STATE;
	}

	@Override
	public Object eGet(int featureID, boolean resolve, boolean coreType) {
		switch (featureID) {
		case FcalPackage.RECEIVE_SFT_INFO_T_R__TRZ_SFT_SIG1_V:
			return getTRZ_SFT_SIG1_V();
		case FcalPackage.RECEIVE_SFT_INFO_T_R__TRZ_SFT_SIG2_V:
			return getTRZ_SFT_SIG2_V();
		case FcalPackage.RECEIVE_SFT_INFO_T_R__TRZ_SFT_DIAG:
			return getTRZ_SFT_DIAG();
		case FcalPackage.RECEIVE_SFT_INFO_T_R__TRZ_SFT_AUTO:
			return getTRZ_SFT_AUTO();
		case FcalPackage.RECEIVE_SFT_INFO_T_R__TRZ_SFT_STATE:
			return getTRZ_SFT_STATE();
		}
		return super.eGet(featureID, resolve, coreType);
	}

	@Override
	public void eSet(int featureID, Object newValue) {
		switch (featureID) {
		case FcalPackage.RECEIVE_SFT_INFO_T_R__TRZ_SFT_SIG1_V:
			setTRZ_SFT_SIG1_V((Float) newValue);
			return;
		case FcalPackage.RECEIVE_SFT_INFO_T_R__TRZ_SFT_SIG2_V:
			setTRZ_SFT_SIG2_V((Float) newValue);
			return;
		case FcalPackage.RECEIVE_SFT_INFO_T_R__TRZ_SFT_DIAG:
			setTRZ_SFT_DIAG((Integer) newValue);
			return;
		case FcalPackage.RECEIVE_SFT_INFO_T_R__TRZ_SFT_AUTO:
			setTRZ_SFT_AUTO((Integer) newValue);
			return;
		case FcalPackage.RECEIVE_SFT_INFO_T_R__TRZ_SFT_STATE:
			setTRZ_SFT_STATE((Integer) newValue);
			return;
		}
		super.eSet(featureID, newValue);
	}

	@Override
	public void eUnset(int featureID) {
		switch (featureID) {
		case FcalPackage.RECEIVE_SFT_INFO_T_R__TRZ_SFT_SIG1_V:
			setTRZ_SFT_SIG1_V(0.0f);
			return;
		case FcalPackage.RECEIVE_SFT_INFO_T_R__TRZ_SFT_SIG2_V:
			setTRZ_SFT_SIG2_V(0.0f);
			return;
		case FcalPackage.RECEIVE_SFT_INFO_T_R__TRZ_SFT_DIAG:
			setTRZ_SFT_DIAG(0);
			return;
		case FcalPackage.RECEIVE_SFT_INFO_T_R__TRZ_SFT_AUTO:
			setTRZ_SFT_AUTO(0);
			return;
		case FcalPackage.RECEIVE_SFT_INFO_T_R__TRZ_SFT_STATE:
			setTRZ_SFT_STATE(0);
			return;
		}
		super.eUnset(featureID);
	}

	@Override
	public boolean eIsSet(int featureID) {
		switch (featureID) {
		case FcalPackage.RECEIVE_SFT_INFO_T_R__TRZ_SFT_SIG1_V:
			return tRZ_SFT_SIG1_V != 0.0f;
		case FcalPackage.RECEIVE_SFT_INFO_T_R__TRZ_SFT_SIG2_V:
			return tRZ_SFT_SIG2_V != 0.0f;
		case FcalPackage.RECEIVE_SFT_INFO_T_R__TRZ_SFT_DIAG:
			return tRZ_SFT_DIAG != 0;
		case FcalPackage.RECEIVE_SFT_INFO_T_R__TRZ_SFT_AUTO:
			return tRZ_SFT_AUTO != 0;
		case FcalPackage.RECEIVE_SFT_INFO_T_R__TRZ_SFT_STATE:
			return tRZ_SFT_STATE != 0;
		}
		return super.eIsSet(featureID);
	}
} //RECEIVE_SFT_INFO_T_R
